deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/WriteRelativeToAttributeHandler.java

```diff
@@ -0,0 +1,87 @@
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.server.deployment.scanner;
+
+import java.io.File;
+
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.ReloadRequiredWriteAttributeHandler;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.controller.services.path.PathManager;
+import org.jboss.as.controller.services.path.PathManagerService;
+import org.jboss.as.server.deployment.scanner.logging.DeploymentScannerLogger;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceController;
+
+/**
+ * @author wangc
+ *
+ */
+public class WriteRelativeToAttributeHandler extends ReloadRequiredWriteAttributeHandler {
+
+    static final WriteRelativeToAttributeHandler INSTANCE = new WriteRelativeToAttributeHandler();
+
+    public WriteRelativeToAttributeHandler() {
+        super(DeploymentScannerDefinition.RELATIVE_TO);
+    }
+
+    @Override
+    protected void validateUpdatedModel(final OperationContext context, final Resource resource)
+            throws OperationFailedException {
+
+        final ModelNode model = resource.getModel();
+
+        ModelNode path = model.get(CommonAttributes.PATH);
+
+        final ServiceController<PathManager> controller = (ServiceController<PathManager>) context.getServiceRegistry(false)
+                .getService(PathManagerService.SERVICE_NAME);
+        if (controller != null) {
+            if (model.hasDefined(CommonAttributes.RELATIVE_TO)) {
+                final ModelNode relativeToPath = model.get(CommonAttributes.RELATIVE_TO);
+                String pathName;
+                try {
+                    pathName = controller.getValue().resolveRelativePathEntry(path.asString(), relativeToPath.asString());
+                } catch (IllegalArgumentException e) {
+                    throw new OperationFailedException(
+                            DeploymentScannerLogger.ROOT_LOGGER.pathEntryNotFound(relativeToPath.asString()));
+                }
+                if (pathName != null) {
+                    File relativePath = new File(pathName);
+                    if (!relativePath.exists()) {
+                        throw new OperationFailedException(
+                                DeploymentScannerLogger.ROOT_LOGGER.directoryIsNonexistent(relativePath.getAbsolutePath()));
+                    }
+                    if (!relativePath.isDirectory()) {
+                        throw new OperationFailedException(
+                                DeploymentScannerLogger.ROOT_LOGGER.isNotADirectory(relativePath.getAbsolutePath()));
+                    }
+                    if (!relativePath.canWrite()) {
+                        throw new OperationFailedException(
+                                DeploymentScannerLogger.ROOT_LOGGER.directoryIsNotWritable(relativePath.getAbsolutePath()));
+                    }
+                }
```

Comment:
This block starting at L69 doesn't belong in this class. It's not the task of the deployment-scanner subsystem to validate these, beyond ensuring the attribute value is a reference is to a known path. Validation of the full path in WritePathAttributeHandler covers the deployment-scanner subsystem's needs.